### PR TITLE
Use of hard-coded password is a bad practice

### PR DESCRIPTION
**Pull request by akondasif**
Greetings, 

I am a security researcher, who is looking for security smells in Puppet scripts. 
I noticed instances of hard-coded passwords, which are against the best practices 
recommended by Common Weakness Enumeration (CWE) [https://cwe.mitre.org/data/definitions/259.html] and also by other security practitioners.
I have added hiera support to mitigate this smell. Feedback is welcome. 

Here is where I noticed hard-coded passwords: https://github.com/justindowning/puppet-statsd/blob/master/manifests/params.pp

### CODE REVIEW

File: hiera.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+:hierarchy:
+  - common
+:backends:
+  - yaml
+:yaml:
+:datadir: 'hieradata'
```
Comment (b4ldr) [inline hiera.yaml]: this should be version5 format e.g.

```yaml
version: 5
defaults:
  datadir: 'hieradata'
  data_hash: 'yaml_data'
hierarchy:
  - name: "common"
    path: "common.yaml"
```

File: hieradata/common.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+---
+influx_db_pwd: 'root'
```
Comment (b4ldr) [inline hieradata/common.yaml]: this should be named `statsd::influxdb_password ` to make use of automatic data lookup

File: manifests/params.pp [hunk 3]
```diff
@@ -50,7 +50,7 @@
   $influxdb_port                     = '8086'
   $influxdb_database                 = 'statsd'
   $influxdb_username                 = 'root'
-  $influxdb_password                 = 'root'
+  $influxdb_password                 = hiera('influx_db_pwd')
```
Comment (b4ldr) [inline manifests/params.pp]: dont use the `hiera` function here either use automatic parameter lookup which requires at least puppet 4.10.  or inherit from the params file

**Review by b4ldr — COMMENTED** [review]
@akondasif this patch doesn't change the security in anyway, it just changes where the cleartext password is stored.   Further a change to use hiera as the backend should be one that moves all parameters to hiera and removes the params file all together.

File: manifests/params.pp (hunk 3 above)
Comment (akondasif) [reply manifests/params.pp]: Thanks for the feedback. I have updated my changes.

**Comment by jdowning** [comment]
As noted by @b4ldr, this change does not really improve the security of the module. Closing.

**Pull request closed without merging**